src/Analysis/Ast/Impl/Analyzer/PythonAnalyzerSession.cs

```diff
@@ -143,13 +145,37 @@ internal sealed class PythonAnalyzerSession {
                     _progress.ReportRemaining(walker.Remaining);
                 }
 
+                var elapsed = stopWatch.Elapsed.TotalMilliseconds;
```

Comment:
Can we move it outside of the `finally` block and put into its individual method?

Reply:
What do you mean? This needs to be accessed after `stopWatch.Stop()`, right?

Reply:
It will be stopped after `finally`. Code in `finally` blocks is always executed (even in case of `Thread.Abort`), so I'd prefer not to call heavy methods from it.

Reply:
Oh, you meant to move it below, not above, I see.

Reply:
I moved things outward; let me know if that's what you wanted.

Reply:
Yep. And move that code into separate methods, something like `SendTelemetry` & `LogResults`.

Reply:
I moved them out, including their checks. I can move the if statements back out, if you'd prefer.